src/dotnet/CarbonAware.Tools.WattTimeClient/IWattTimeClient.cs

```diff
@@ -0,0 +1,99 @@
+{
+    /// <summary>
+    /// Async method to get observed emission data for a given balancing authority and time period.
+    /// </summary>
+    /// <param name="balancingAuthorityAbbreviation">Balancing authority abbreviation</param>
+    /// <param name="startTime">Start time of the time period</param>
+    /// <param name="endTime">End time of the time period</param>
+    /// <returns>An <see cref="Task{IEnumerable}{GridEmissionDataPoint}"/> which contains all emissions data points in a period.</returns>
+    /// <exception cref="WattTimeClientException">Can be thrown when errors occur connecting to WattTime client.  See the WattTimeClientException class for documentation of expected status codes.</exception>
+    public Task<IEnumerable<GridEmissionDataPoint>> GetDataAsync(string balancingAuthorityAbbreviation, string startTime, string endTime);
+
+    /// <summary>
+    /// Async method to get observed emission data for a given balancing authority and time period.
+    /// </summary>
+    /// <param name="balancingAuthority">Balancing authority</param>
+    /// <param name="startTime">Start time of the time period</param>
+    /// <param name="endTime">End time of the time period</param>
+    /// <returns>An <see cref="Task{IEnumerable}{GridEmissionDataPoint}"/> which contains all emissions data points in a period.</returns>
+    /// <exception cref="WattTimeClientException">Can be thrown when errors occur connecting to WattTime client.  See the WattTimeClientException class for documentation of expected status codes.</exception>
+    public Task<IEnumerable<GridEmissionDataPoint>> GetDataAsync(BalancingAuthority balancingAuthority, string startTime, string endTime);
+
+    /// <summary>
+    /// Async method to get the most recent 24 hour forecasted emission data for a given balancing authority.
+    /// </summary>
+    /// <param name="balancingAuthorityAbbreviation">Balancing authority abbreviation</param>
+    /// <returns>An <see cref="Task{Forecast}"/> which contains forecasted emissions data points.</returns>
+    /// <exception cref="WattTimeClientException">Can be thrown when errors occur connecting to WattTime client.  See the WattTimeClientException class for documentation of expected status codes.</exception>
+    public Task<Forecast?> GetCurrentForecastAsync(string balancingAuthorityAbbreviation);
+
+    /// <summary>
+    /// Async method to get the most recent 24 hour forecasted emission data for a given balancing authority.
+    /// </summary>
+    /// <param name="balancingAuthority">Balancing authority</param>
+    /// <returns>An <see cref="Task{Forecast}"/> which contains forecasted emissions data points.</returns>
+    /// <exception cref="WattTimeClientException">Can be thrown when errors occur connecting to WattTime client.  See the WattTimeClientException class for documentation of expected status codes.</exception>
+    public Task<Forecast?> GetCurrentForecastAsync(BalancingAuthority balancingAuthority);
```

Comment:
>public Task<Forecast?> GetCurrentForecastAsync(BalancingAuthority balancingAuthority); [](http://example.com/codeflow?start=4&length=86)

based primarily on the existing CA-API Controller approach, I'd also suggest the 2 Get Current Forecast APIs also return an IEnumerable<Forcast>

Reply:
I think it makes sense to leave this as-is.  Anyone calling this function will expect a single item, which is also the most common use-case and explicitly matches the API response.